web/src/saved_replies.ts

```diff
@@ -0,0 +1,157 @@
+import $ from "jquery";
+import assert from "minimalistic-assert";
+import type * as tippy from "tippy.js";
+
+import render_add_saved_reply_modal from "../templates/add_saved_reply_modal.hbs";
+
+import * as channel from "./channel";
+import * as compose_ui from "./compose_ui";
+import * as dialog_widget from "./dialog_widget";
+import * as dropdown_widget from "./dropdown_widget";
+import type {Option} from "./dropdown_widget";
+import {$t, $t_html} from "./i18n";
+import * as keydown_util from "./keydown_util";
+import type {StateData} from "./state_data";
+
+export type SavedReplies = {
+    id: number;
+    title: string;
+    content: string;
+    rendered_content: string;
+    date_created: number;
+};
+
+const ADD_SAVED_REPLY_OPTION_ID = -1;
+let my_saved_replies: SavedReplies[] = [];
+
+export function set_saved_replies(saved_replies: SavedReplies[]): void {
+    my_saved_replies = saved_replies;
+    my_saved_replies.sort((a, b) => a.title.toLowerCase().localeCompare(b.title.toLowerCase()));
```

Comment:
It may be fine to exempt `saved_replies.ts` from test coverage, but I guess it would be better if we write some tests that verify the state of the `my_saved_replies` after performing each of the functions like  `remove_saved_reply`, `add_saved_reply`, etc.

I guess you could have a look at `stream_data.test.js` as an example for `get_options_for_dropdown_widget` test?

Reply:
Added tests for `add_saved_reply`, `remove_saved_reply` and `get_options_for_dropdown_widget`.